heron/metricscachemgr/src/java/com/twitter/heron/metricscachemgr/MetricsCacheManagerServer.java

```
@@ -0,0 +1,94 @@
+import com.twitter.heron.common.network.HeronServer;
+import com.twitter.heron.common.network.HeronSocketOptions;
+import com.twitter.heron.common.network.REQID;
+import com.twitter.heron.metricscachemgr.metricscache.MetricsCache;
+import com.twitter.heron.metricsmgr.MetricsManagerServer;
+import com.twitter.heron.proto.tmaster.TopologyMaster;
+
+/**
+ * server to accept metrics from a particular sink in metrics manager
+ */
+public class MetricsCacheManagerServer extends HeronServer {
+  private static final Logger LOG = Logger.getLogger(MetricsManagerServer.class.getName());
+
+  private MetricsCache metricsCache = null;
+
+  /**
+   * Constructor
+   *
+   * @param s the NIOLooper bind with this socket server
+   * @param host the host of remote endpoint to communicate with
+   * @param port the port of remote endpoint to communicate with
+   */
+  public MetricsCacheManagerServer(NIOLooper s, String host, int port, HeronSocketOptions options,
+                                   MetricsCache cache) {
+    super(s, host, port, options);
+
+    metricsCache = cache;
+  }
+
+  @Override
+  public void onConnect(SocketChannel channel) {
+    LOG.info("MetricsCacheManagerServer onConnect from host:port "
+        + channel.socket().getRemoteSocketAddress());
+  }
+
+  @Override
+  public void onRequest(REQID rid, SocketChannel channel, Message request) {
+    LOG.info("MetricsCacheManagerServer onRequest from host:port "
+        + channel.socket().getRemoteSocketAddress());
+
+    if (request instanceof TopologyMaster.MetricRequest) {
+      LOG.info("received request " + (TopologyMaster.MetricRequest) request);
+      TopologyMaster.MetricResponse resp =
+          metricsCache.GetMetrics((TopologyMaster.MetricRequest) request);
+      LOG.info("query finished, to send response");
+      sendResponse(rid, channel, resp);
+      LOG.info("queued response size " + resp.getSerializedSize());
+    } else {
+      LOG.severe("Unknown kind of request received");
```

would be better to throw an exception than to silently fail. Also please include more helpful info in the error message, like the request and the remote address. Same for other checks in this class

---

the method signature is inherited. the exception is not supported.
besides, a server receiving an unknown request/message is expected rather than exception.

done: add request/message and remote address to log message.